created TLP classes as defined in spec

anlklsim opened the pull request:
The STIX spec defines the exact ID and created date for the 4 TLP markings.  I defined them in a class as static values and created a test Spec.  It's interesting that the markings do not need to be within the bundle.

Note that the Stix Mocking for TLP Marking defs makes up custom properties, but they are not really allowed for the TLP markings.

StephenOTT commented:
Great stuff.

I am waiting for a response on this: https://github.com/oasis-tcs/cti-stix2/issues/152 before merging.

The extreme limitation on the TLPs is very weird, and would like to document its usage before we implement the rest of the TLPs

Further tasks:

- [ ] Update the TLPs to remove custom props
- [ ] Add a config override to allow a implementation to allow custom parsing of TLPs without having to create custom objects...
- [ ] Add logic on deserialization to validate against the new Tlps class that @anlklsim has provided
- [ ] Add some helper logic on object build to validate that the TLP is matched to the spec limitations/requirements.

StephenOTT commented on src/main/java/io/digitalstate/stix/datamarkings/objects/Tlps.java:
```diff
@@ -0,0 +1,80 @@
+/**
```
Remove extra comments

StephenOTT commented on src/main/java/io/digitalstate/stix/datamarkings/objects/Tlps.java:
```diff
@@ -0,0 +1,80 @@
+        MarkingDefinition.Builder builder = MarkingDefinition.builder()
+        		.id("marking-definition--613f2e26-407d-48c7-9eca-b8e91df99dc9")
+        		.definitionType("tlp")
+        		.created(new StixInstant(ZonedDateTime.parse("2017-01-20T00:00:00.000Z", DateTimeFormatter.ISO_DATE_TIME).toInstant(),3))
```
Use the StixInstant.parse(...) static method rather than the ZonedDateTime

StephenOTT replied:
Update each of the objects with the .parse

StephenOTT commented on src/main/java/io/digitalstate/stix/datamarkings/objects/Tlps.java:
```diff
@@ -0,0 +1,80 @@
+	private static MarkingDefinition getTlpWhiteMD() {
+        MarkingDefinition.Builder builder = MarkingDefinition.builder()
+        		.id("marking-definition--613f2e26-407d-48c7-9eca-b8e91df99dc9")
+        		.definitionType("tlp")
```
Move the definitionType value into a static final class var

StephenOTT commented on src/test/groovy/stix/datamarkings/TLPmarkingsSpec.groovy:
```diff
@@ -0,0 +1,96 @@
+import spock.lang.Shared
+import spock.lang.Specification
+
+/**
```
- [ ] review usage of author javadoc tag as its starting to be deprecated in tooling like apache and gradle

StephenOTT commented on src/test/groovy/stix/datamarkings/TLPmarkingsSpec.groovy:
```diff
@@ -0,0 +1,96 @@
+            println "Original Json: ${originalJsonString}"
+
+        then: "Parse Json back into Marking Definition Object"
+            MarkingDefinition parsedMarkingDefinition = (MarkingDefinition)StixParsers.parseObject(originalJsonString)
```
- [ ] Update mock Marking Def to remove the custom props on the TLPs as per @anlklsim comments

StephenOTT replied:
- [ ] Add usage of the Generics parser StixParser.parse() for testing

StephenOTT commented on src/test/groovy/stix/datamarkings/TLPmarkingsSpec.groovy:
```diff
@@ -0,0 +1,96 @@
+		MarkingDefinition green = Tlps.TLP_GREEN;
+		StixInstant now = new StixInstant();
+		
+		Indicator ind = Indicator.builder()
```
Lets move this into a json file similar to the Bundle tests.  The current test is basically a duplicate of the mock generation test.  The json file tests is a "hardened" data example where the json file represents the "pure" unchanging json to test against

StephenOTT commented:
@anlklsim what scenario causes the NPE that you fixed for the getCause() ?

anlklsim commented:
I had mistakenly called STIXParsers.parseObject on my bundle in my test, which threw an IOExeption in the method (since it was not a BundleableObject), and the if statement threw an NPE on ex.getClause() and therefore only the NPE was logged in the Junit as the error (hiding the true exception message).  Stepping thru the debugger, I saw the error, added the null check, and then the true error printed out.  Only then, did I notice that I was calling the wrong method.  I added the same check to parseBundle to be complete.
